CMakeLists.txt

```diff
@@ -26,6 +26,16 @@ if (BUILD_COMMAND_LINE_TOOLS)
   add_subdirectory(dump-pe)
 endif ()
 
+# `format` target.
+file(GLOB_RECURSE PEPARSE_ALL_SOURCES *.cpp *.h)
+add_custom_target(
+  format
+  COMMAND clang-format -i -style=file ${PEPARSE_ALL_SOURCES}
```

Comment:
Minor nit: When installing `clang-format` tool on Ubuntu, if you install a versioned one (not just `apt install clang-format`), it will show up suffixed with the version number: `apt install clang-format-9` will be `clang-format-9` invocation.

If we wanted people to use an exact version, we should have a version check here or call it by `clang-format-9` (not sure if other systems also use version suffix).

In the end, the CI system will tell them that their linter is probably the wrong version if there are any differences, so we don't _need_ to fix this now, if there isn't a nice or easy way to do the version check.

Reply:
Yeah, I constantly run into this 😅 

The problem here is that Debian/Ubuntu themselves aren't consistent about defaults -- some of the packages for `clang-format-N` will always alias themselves to `clang-format`, others always won't, and yet others will only if there isn't another already aliased to it. Similarly for the "universe" packages vs. the ones created by the LLVM dev team.

I think what we should *really* be doing is using a custom GitHub Action that allows the version and settings for `clang-format` to be fed in, and dedupe all of our formatting boilerplate across all of our projects. That still makes things confusing when people's local `clang-format` is a mismatch, but at least we'll be explicit and consistent on the CI about what we're expecting.

Reply:
> using a custom GitHub Action that allows the version and settings for clang-format to be fed in

I keep wanting to do something like this, but

> That still makes things confusing when people's local `clang-format` is a mismatch

is the thing I get stuck up on whether it's worth it. I also feel like docker is a bit overkill for just doing a local formatting run, and then you have to wrestle with Docker mount permissions too. Who cares if it's docker under the hood for CI though. I might work on it anyway.

Would be nice if the `.clang-format` file had the option to check the version of the tool that parses it and enforce it that way maybe? 🤷

Reply:
> Would be nice if the `.clang-format` file had the option to check the version of the tool that parses it and enforce it that way maybe? 🤷

Sounds like a good patch to send 😉